src/pages/workspaces/workspace/analysis/modals/compute-modal-components.ts

```diff
@@ -0,0 +1,109 @@
+// eslint-disable-next-line lodash-fp/use-fp
+import { CurriedFunction1 } from 'lodash'
+import { div, h, label } from 'react-hyperscript-helpers'
+import { Link, Select } from 'src/components/common'
+import { NumberInput } from 'src/components/input'
+import TooltipTrigger from 'src/components/TooltipTrigger'
+import { useUniqueId } from 'src/libs/react-utils'
+import { computeStyles } from 'src/pages/workspaces/workspace/analysis/modals/modalStyles'
+import { pdTypes } from 'src/pages/workspaces/workspace/analysis/runtime-utils'
+
+
+interface IComputeConfig {
+  selectedPersistentDiskSize: number
+  selectedPersistentDiskType: {
+    label: string
+    displayName: string
+    regionToPricesName: string
+  }
+  masterMachineType: any
+  masterDiskSize: number
+  numberOfWorkers: number
+  numberOfPreemptibleWorkers: number
+  workerMachineType: string
+  workerDiskSize: number
+  componentGatewayEnabled: boolean
+  gpuEnabled: boolean
+  hasGpu: boolean
+  gpuType: string
+  numGpus: number
+  autopauseThreshold: number
+  computeRegion: string
+  computeZone: string
+}
+
+interface PersistentDiskProps {
+  diskExists: boolean
+  computeConfig: IComputeConfig
+  updateComputeConfig: CurriedFunction1<string, void>
+  handleLearnMoreAboutPersistentDisk: React.MouseEventHandler
+}
+
+interface PersistentDiskTypeProps {
+  diskExists: boolean
+  computeConfig: IComputeConfig
+  updateComputeConfig: CurriedFunction1<string, void>
```

Comment:
Does this type fully capture the desired signature? We want a function that takes a string and returns a function that accepts the value from Select's `onChange`. It looks like this has a type parameter only for the string, not the selected value?

Reply:
can you clarify this a bit? I'm kind of confused

Reply:
Admittedly, I don't fully understand the `CurriedFunction1` type, so I'm guessing based on the single type parameter.

It looks like this may be effectively:
```ts
(arg: string) => () => void
```

When we want:
```ts
(arg: string) => (diskType: string) => void
```

Reply:
ok thanks for clarifying